internal/dag/policy.go

```diff
@@ -304,6 +305,53 @@ func escapeHeaderValue(value string, dynamicHeaders map[string]string) string {
 	return escapedValue
 }
 
+func cookieRewritePolicies(policies []contour_api_v1.CookieRewritePolicy) ([]CookieRewritePolicy, error) {
+	validPolicies := make([]CookieRewritePolicy, 0, len(policies))
+	cookieNames := map[string]struct{}{}
+	for _, p := range policies {
+		if _, exists := cookieNames[p.Name]; exists {
+			return nil, fmt.Errorf("duplicate cookie rewrite rule for cookie %q", p.Name)
+		}
+		cookieNames[p.Name] = struct{}{}
+		policiesSet := 0
+		var path *string
+		if p.PathRewrite != nil {
+			policiesSet++
+			path = pointer.String(p.PathRewrite.Value)
+		}
+		var domain *string
+		if p.DomainRewrite != nil {
+			policiesSet++
+			domain = pointer.String(p.DomainRewrite.Value)
+		}
+		secure := uint(0)
+		if p.Secure != nil {
+			policiesSet++
+			secure++
+			if *p.Secure {
+				secure++
+			}
+		}
```

Comment:
Integer `secure` "security level" might be bit difficult to understand without having a comment close by. It was well explained in `dag.go` 
```
// Using an uint since pointer to boolean gets dereferenced in golang
// text templates so we have no way of distinguishing if unset or set to false.
// 0 means unset, 1 means false, 2 means true
```

Reply:
Might be able to get rid of this since I’m refactoring a bit, but will comment more here if it stays

Reply:
+1 on comment if it stays.